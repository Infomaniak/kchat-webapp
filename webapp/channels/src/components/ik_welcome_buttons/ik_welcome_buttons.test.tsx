import {fireEvent, screen} from '@testing-library/react';
import React from 'react';

import type EmojiMap from 'utils/emoji_map';

import {renderWithContext} from 'tests/react_testing_utils';

import IkWelcomeButtons from './ik_welcome_buttons';

jest.mock('mattermost-redux/utils/emoji_utils', () => {
    return {
        getEmojiImageUrl: jest.fn(() => ''),
    };
});

describe('components/IkWelcomeButtons', () => {
    const baseProps = {
        actions: {
            openModal: jest.fn(),
        },
        emojiMap: new Map() as unknown as EmojiMap,
    };

    test('should match snapshot', () => {
        const {container} = renderWithContext(
            <IkWelcomeButtons {...baseProps}/>,
        );
        expect(container).toMatchSnapshot();
    });

    test('should open DM modal when clicking "Write to someone"', () => {
        renderWithContext(
            <IkWelcomeButtons {...baseProps}/>,
        );

        const writeButton = screen.getByText(/Write to someone/i);
        fireEvent.click(writeButton);

        expect(baseProps.actions.openModal).toHaveBeenCalledWith(expect.objectContaining({
            modalId: 'create_dm_channel',
            dialogProps: {isExistingChannel: false},
        }));
    });

    test('should open Quick Switch modal when clicking "Browse channels"', () => {
        renderWithContext(
            <IkWelcomeButtons {...baseProps}/>,
        );

        const browseButton = screen.getByText(/Browse channels/i);
        fireEvent.click(browseButton);

        expect(baseProps.actions.openModal).toHaveBeenCalledWith(expect.objectContaining({
            modalId: 'quick_switch',
        }));
    });

    test('should open apps link in new tab when clicking "Download the apps"', () => {
        const windowOpenSpy = jest.spyOn(window, 'open').mockImplementation();

        renderWithContext(
            <IkWelcomeButtons {...baseProps}/>,
        );

        const downloadButton = screen.getByText(/Download the apps/i);
        fireEvent.click(downloadButton);

        expect(windowOpenSpy).toHaveBeenCalledWith(
            'https://infomaniak.com/gtl/apps.kchat',
            '_blank',
            'noopener,noreferrer',
        );

        windowOpenSpy.mockRestore();
    });
});
