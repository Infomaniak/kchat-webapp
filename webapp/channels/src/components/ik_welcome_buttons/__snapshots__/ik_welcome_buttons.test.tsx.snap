// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/IkWelcomeButtons should match snapshot 1`] = `
<div>
  <div
    class="system-bot-message"
  >
    <p>
      Hello! Happy to see you here and welcome 
      <span
        class="emoticon emoticon--blush"
      />
      <br />
      I'm here to answer your questions and send you reminders and notifications about your kSuite products when needed.
    </p>
    <p
      class="mt-2"
    >
      <b>
        To get started
      </b>
      , here are some quick actions to explore kChat
    </p>
    <div
      class="system-bot-message__buttons"
    >
      <button
        class="btn btn-sm btn-outline"
        type="button"
      >
        <span
          class="emoticon emoticon--writing_hand"
        />
         Write to someone
      </button>
      <button
        class="btn btn-sm btn-outline"
        type="button"
      >
        <span
          class="emoticon emoticon--mag"
        />
         Browse channels
      </button>
      <button
        class="btn btn-sm btn-outline"
        type="button"
      >
        <span>
          <span
            class="emoticon emoticon--computer"
          />
          <span
            class="emoticon emoticon--iphone"
          />
        </span>
         Download the apps
      </button>
    </div>
    <p
      class="mt-1 mb-2"
    >
      <span
        class="emoticon emoticon--arrow_down"
      />
       Or 
      <b>
        ask me a question
      </b>
       directly here!
    </p>
  </div>
</div>
`;
